src/main/java/com/microsoft/cse/helium/app/config/CosmosDbConfig.java

```diff
@@ -39,16 +43,24 @@ public CosmosDbConfig(IConfigurationService configService) {
   @Bean
   @Primary
   public CosmosDBConfig buildCosmosDbConfig() {
-    String uri = configurationService.getConfigEntries().get(Constants.COSMOS_URL_KEYNAME);
-    String key = configurationService.getConfigEntries().get(Constants.COSMOS_KEY_KEYNAME);
-    String dbName = configurationService.getConfigEntries().get(Constants.COSMOS_DATABASE_KEYNAME);
-
-    ConnectionPolicy policy = new ConnectionPolicy();
-    RetryOptions retryOptions = new RetryOptions();
-    retryOptions.maxRetryWaitTimeInSeconds(60);
-    policy.retryOptions(retryOptions);
-    return CosmosDBConfig.builder(uri, key, dbName)
+    try {
+
+      String uri = configurationService.getConfigEntries().get(Constants.COSMOS_URL_KEYNAME);
+      String key = configurationService.getConfigEntries().get(Constants.COSMOS_KEY_KEYNAME);
+      String dbName = configurationService.getConfigEntries()
+          .get(Constants.COSMOS_DATABASE_KEYNAME);
+
+      ConnectionPolicy policy = new ConnectionPolicy();
+      RetryOptions retryOptions = new RetryOptions();
+      retryOptions.maxRetryWaitTimeInSeconds(60);
+      policy.retryOptions(retryOptions);
+
+      return CosmosDBConfig.builder(uri, key, dbName)
         .requestOptions(requestOptions).connectionPolicy(policy)
         .build();
+    } catch (Exception ex) {
+      logger.error("buildCosmosDbConfig failed with error: " + ex.getMessage());
+      return null;
```

Comment:
We are mixing two different types of error handling here. We are catching the exception like java and then returning null (to be caught) like non-exception programming.  You should throw the exception from here and let the program crash with the appropriate error message.

Reply:
It now throws.